File: backend/src/main/java/pl/edu/agh/utp/records/TransactionsGraph.java
package pl.edu.agh.utp.records;

import java.util.List;
import java.util.UUID;

public record TransactionsGraph(List<Vertex> vertices, List<Edge> edges) {

    public record Vertex(UUID userId, String username) {}

    public record Edge(UUID from, UUID to, double value) {}

    public record Pair(UUID from, UUID to) {}

}

File: frontend/app/src/main/java/pl/edu/agh/utp/model/graph/TransactionsGraph.kt
package pl.edu.agh.utp.model.graph

import java.util.UUID

data class TransactionsGraph(val vertices: List<Vertex>, val edges: List<Edge>) {

    data class Vertex(val userId: UUID, val username: String)

    data class Edge(val from: UUID, val to: UUID, val value: Double)

    fun toNodesString(): String {
        val nodesList = vertices.map { vertex ->
            "{ id: '${vertex.userId}', label: '${vertex.username}' }"
        }
        return nodesList.joinToString(",\n", "[\n", "\n]")
    }

    fun toEdgesString(): String {
        val edgesList = edges.map { edge ->
            "{ from: '${edge.from}', to: '${edge.to}', label: '${edge.value}' }"
        }
        return edgesList.joinToString(",\n", "[\n", "\n]")
    }
}